src/components/OperatorChecklist.tsx

```diff
@@ -0,0 +1,289 @@
+import { TABLET } from '~/shared/utils/styled'
+import { StatCellLabelTip } from '~/components/StatGrid'
+import { useInterceptHeartbeats } from '~/hooks/useInterceptHeartbeats'
+import { SponsorshipPaymentTokenName } from './SponsorshipPaymentTokenName'
+
+export function OperatorChecklist({ operatorId }: { operatorId: string | undefined }) {
+    const { funded, nodesDeclared, nodesFunded, nodesReachable, nodesRunning } =
+        useOperatorChecklist(operatorId)
+
+    return (
+        <>
+            <ChecklistItem
+                state={funded}
+                tip={
+                    <p>
+                        You need to deposit <SponsorshipPaymentTokenName /> tokens into
+                        your operator contract before you can stake on sponsorships or
+                        receive delegations.
+                    </p>
+                }
+            >
+                Operator funded
+            </ChecklistItem>
+            <Separator />
+            <ChecklistItem
+                state={nodesDeclared}
+                tip={
+                    <p>
+                        Configure your nodes with private keys and declare the
+                        corresponding addresses to the operator contract.
+                    </p>
+                }
+            >
+                Node addresses declared
+            </ChecklistItem>
+            <Separator />
+            <ChecklistItem
+                state={nodesFunded}
+                tip={
+                    <p>
+                        Your node addresses must be funded with some MATIC for Polygon
+                        transaction fees. This alert triggers if a balance is less than
+                        0.1 MATIC.
+                    </p>
+                }
+            >
+                Node addresses funded
+            </ChecklistItem>
+            <Separator />
+            <ChecklistItem
+                state={nodesRunning}
+                tip={
+                    <p>
+                        You need to run Streamr nodes with the operator plugin enabled.
+                        The nodes will do the work you sign up for when you stake on
+                        sponsorships.
+                    </p>
+                }
+            >
+                Nodes running
+            </ChecklistItem>
+            <Separator />
+            <ChecklistItem
+                disabled
+                state={nodesReachable}
+                tip={<p>The websocket port on your nodes should be reachable.</p>}
+            >
+                Nodes reachable
+            </ChecklistItem>
+        </>
+    )
+}
+
+interface OperatorChecklist {
+    funded?: boolean | undefined
+    nodesDeclared?: boolean | undefined
+    nodesFunded?: boolean | undefined
+    nodesReachable?: boolean | undefined
+    nodesRunning?: boolean | undefined
+}
+
+function useOperatorChecklist(operatorId: string | undefined): OperatorChecklist {
```

Comment:
I prefer if we keep hooks in the hooks folder so that we all know where to look for them :)